File: ShowOtherMasters/ShowOtherMasters.m
#import <Foundation/Foundation.h>
#import <GlyphsCore/GSLayer.h>
#import <GlyphsCore/GSGlyph.h>
#import <GlyphsCore/GSComponent.h>
#import <GlyphsCore/GlyphsReporterProtocol.h>
#import <GlyphsCore/GSGlyphViewControllerProtocol.h>

#import "ShowOtherMasters.h"

@interface ShowOtherMasters ()
@property (nullable) NSViewController * _controller;
@end

@implementation ShowOtherMasters

- (id)init {
    self = [super init];
    return self;
}

- (NSUInteger)interfaceVersion {
    return 1;
}

- (NSString *)title {
    NSString *aString = [[NSBundle bundleForClass:[self class]] localizedStringForKey:@"OtherMasters" value:@"" table:nil];
    return aString ? aString : @"Other";
}

- (NSString *)keyEquivalent {
    return @"`";
}

- (int)modifierMask {
    return NSCommandKeyMask;
}

- (void)setController:(NSViewController <GSGlyphEditViewControllerProtocol> *)controller {
    self._controller = controller;
}

- (void)drawForegroundForLayer:(GSLayer *)layer {
    [[NSColor colorWithCalibratedWhite:0.5 alpha:0.2] set];
    if (layer && layer.parent && layer.parent.layers) {
        NSDictionary *allLayers = layer.parent.layers;
        for (NSObject *aLayer in [allLayers allValues]) {
            if ([aLayer isMemberOfClass:[GSLayer class]]) {
                GSLayer *myLayer = (GSLayer *) aLayer;
                if (myLayer.paths && myLayer.paths.count > 0) {
                    if (myLayer.bezierPath) {
                        [myLayer.bezierPath fill];
                    }
                }
                if (myLayer.components && myLayer.components.count > 0) {
                    for (NSObject *aComponent in myLayer.components) {
                        if ([aComponent isMemberOfClass:[GSComponent class]]) {
                            GSComponent *myComponent = (GSComponent *) aComponent;
                            NSBezierPath *myComponentBezierPath = [myComponent bezierPath];
                            if (myComponentBezierPath) {
                                [myComponentBezierPath fill];
                            }
                        }
                    }
                }
            }
        }
    }
}

@end

File: ShowOtherMasters/ShowOtherMasters.swift
//
//  ShowOtherMasters.swift
//  ShowOtherMasters
//
//  Created by Ben Syverson on 7/1/14.
//  Copyright (c) 2014 Ben Syverson. All rights reserved.
//

import Foundation
import Cocoa


class ShowOtherMasters : NSObject, GlyphsReporter {
	init() {
		
	}
	
	var _controller : NSViewController?
	
	func drawBackgroundForLayer(layer: GSLayer) {
		
	}
	
	func interfaceVersion() -> Int
	{
		return 1
	}
	
	func title() -> String!
	{
		if let aString = NSBundle(forClass: ShowOtherMasters.self).localizedStringForKey("OtherMasters", value: "", table: nil) {
			return aString
		}
		return "Other"
	}
	
	func keyEquivalent() -> String!
	{
		return "`"
	}
	
	func modifierMask() -> CUnsignedInt {
		return NSCommandKeyMask
	}
	
	func setController(Controller: NSViewController!)
	{
		_controller = Controller
	}
	
	func drawBackgroundForLayer(Layer: GSLayer?)
	{
		
	}
	
	func drawForegroundForLayer(Layer: GSLayer?)
	{
		NSColor(white:0.5, alpha:0.2).set()
		if let allLayers = Layer?.parent?.layers? {
			for aLayer : AnyObject in allLayers.allValues() {
				if let myLayer = aLayer as? GSLayer {
					if myLayer.paths?.count > 0 {
						if let bezierPath = myLayer.bezierPath as NSBezierPath? {
							bezierPath.fill()
						}
					}
					if myLayer.components?.count > 0 {
						for aComponent : AnyObject in myLayer.components {
							if let myComponent = aComponent as? GSComponent {
								myComponent.bezierPath()?.fill()
							}
						}
					}
				}
			}
		}
	}
	
}

